src/cleanfigure.m

```diff
@@ -672,113 +608,107 @@ function movePointsCloser(meta, handle)
 
   % Get which points are in an extended box (the limits of which
   % don't exceed TeX's memory).
-  dataIsInLargeBox = isInBox(data(:,1:2), ...
-                             largeXLim, largeYLim);
+  dataIsInLargeBox = isInBox(data(:, 1:2), largeXLim, largeYLim);
 
   % Count the NaNs as being inside the box.
-  dataIsInLargeBox = dataIsInLargeBox | any(isnan(data)')';
+  dataIsInLargeBox = dataIsInLargeBox | any(isnan(data), 2);
 
-  % Loop through all points which are to be included in the plot yet do not
-  % fit into the extended box, and gather the points by which they are to be
+  % Find all points which are to be included in the plot yet do not fit
+  % into the extended box, and gather the points by which they are to be
   % replaced.
-
-  replaceIndices = find(~dataIsInLargeBox)';
-  m = length(replaceIndices);
-  r = cell(m, 1);
-  for k = 1:m
-      i = replaceIndices(k);
-      r{k} = [];
-      if i > 1 && all(isfinite(data(i-1,:)))
-          newPoint = moveToBox(data(i,:), data(i-1,:), largeXLim, largeYLim);
-          % Don't bother if the point is inf:
-          % There's no intersection with the large box, so even the
-          % connection between the two after they have been moved
-          % won't be probably be visible.
-          if all(isfinite(newPoint))
-              r{k} = [r{k}; newPoint];
-          end
+  replaceIndices = find(~dataIsInLargeBox);  
+  
+  % Only replace Indices of there are some to replace
+  if (~isempty(replaceIndices))
+      % Get the indices of those segments, where the right point might be moved
+      id_right = replaceIndices(replaceIndices > 1);
+      id_right = id_right(all(isfinite(data(id_right,:)), 2));
+
+      % Get the indices of those segments where the left point might be moved
+      id_left  = replaceIndices(replaceIndices < size(data, 1));
+      id_left  = id_left(all(isfinite(data(id_left,:)), 2));
```

Comment:
Hm, i think it would be best to check depending on the size of replaceIndices.

isFiniteData = all(isfinite(data(replaceIndices,:), 2)); Checking all data might be a huge waste if the data is large and all of it inside the borders